templates/windows/project.nuspec.erb

```diff
@@ -2,7 +2,41 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
   <metadata>
     <id><%= "#{@name}-#{@platform.architecture}" %></id>
-    <version><%= "#{@version}.#{@release}" %></version>
+    <!--
+    Nuget doesn't handle versions well. We have the option of specifying a
+    development release, but that requires a dash followed first by a letter,
+    and then a random string with no punctuation (only letters and numbers).
+    Otherwise, we require at most four elements seperated by period. Each element
+    must only contain numbers. This doesn't play well with our versioning scheme
+    based on git.
+
+    We assume we either have a git tag with three elements or some number of
+    commits beyond a tag, which would take the form of '1.3.2.123.g45e6cd8'. We
+    can't guarantee the sha at the end will begin with a letter. We also can't
+    assume that the sha will only contain numbers. As such, we can't use the short
+    sha from the git version. We then assume the first four elements only contain
+    numbers, and discard the rest.
+
+    We also can't guess how many version elements a project will have. If we're
+    using the traditional semver scheme, we should have three elements to the tag
+    (i.e., '1.3.2'). If this is a development build, we also have available the
+    number of commits beyond the tag (i.e., '123'). As long as all of these values
+    are numeric, we can use them in the version. I'm going to assume that's the
+    case. Since we already have four elements here, we cannot include the release
+    number.
+
+    If the build is from a tag with no additional commits, and there are less then
+    four elements in the version, then we can include the release number.
+
+    Versioning is difficult...
+    -->
+    <%- version_elements = @version.split('.') -%>
+    <%- if version_elements.count >= 4 -%>
+      <%- version = "#{version_elements[0..3].join('.').gsub(/[a-zA-Z]/, '')}-v#{version_elements[4..-1].join('')}" -%>
+    <%- else -%>
+      <%- version = "#{@version}.#{@release}".gsub(/[a-zA-Z]/, '') -%>
+    <%- end -%>
```

Comment:
Not sure where this template is executed, but do you have access to the methods above? This looks like a lot of copy/pasted code